### PR TITLE
Performance tests

### PR DESCRIPTION
**Pull request by n00m4d**
# What
This PR extends ability of testing DAS API with performance tests. Also 3 new endpoints were added.

### CODE REVIEW

File: src/performance_measurement.rs [hunk 1]
```diff
@@ -0,0 +1,314 @@
...
+
+                tokio::time::sleep(tokio::time::Duration::from_millis(100)).await;
+            }
+
+            if self.active {
+                debug!("Worker #{} is sending API request", self.id);
+                let (command, arg_key) = self.keys_fetcher.get_random_command();
+
+                let body = {
+                    if command == GET_ASSET_METHOD {
```
Comment (StanChe) [inline src/performance_measurement.rs]: Why not match here as well?

File: README.md [hunk 2]
```diff
@@ -2,6 +2,8 @@
 
 This repository is dedicated to the DAS-API Testing Suite, designed to facilitate the comparison of responses from two distinct providers, namely the reference and the testing entities.
 
+Also it has performance tests.
```
Comment (StanChe) [inline README.md]:
```suggestion
It also includes the framework to run performance tests for the selected host.
```

File: README.md [hunk 3]
```diff
@@ -13,7 +15,7 @@ export RUST_LOG=info
 ```
 5. Initiate the testing process with the command below, ensuring the path to your configuration file is correctly specified.
 ```bash
-cargo run -- --config-path=/path/to/your/config.json
+cargo run -- --config-path=/path/to/your/config.json ----test-type=integrity/performance
```
Comment (StanChe) [inline README.md]:
```suggestion
cargo run -- --config-path=</path/to/your/config.json> --test-type=<integrity|performance>
```

File: README.md [hunk 4]
```diff
@@ -26,14 +28,20 @@ Within the `config/config_example.json` file located in this repository, you wil
...
 * The `testing_file_path` parameter specifies the local file path containing the test public keys.
 * The `test_retries` parameter determines the number of attempts for each test before it is deemed unsuccessful. Configurations with values less than 1 will result in an error, whereas a value of 1 signifies immediate failure upon the first unsuccessful attempt.
 * The `log_differences` boolean flag controls the logging of discrepancies in failed tests, with a true value enabling this feature.
 * The `difference_filter_regexes` provides an array of regular expressions designed to exclude certain disparities from the comparative analysis of provider responses. This feature is particularly useful for ignoring known, inconsequential differences.
+* The `num_of_virtual_users` parameter specifies the number of threads that will send requests in parallel mode to the API. **For performance test only**
+* The `test_duration_time` parameter specifies the duration, in seconds, for which the test will run. **For performance test only**
+
+For performance tests `testing_host` API will be used.
```
Comment (StanChe) [inline README.md:43]: Can we also include a note on the load created during the perf test? So the user understands whether she'll exhaust the limits of the provider.

**Review by StanChe — APPROVED** [review]

File: src/api.rs [hunk 5]
```diff
@@ -1,6 +1,8 @@
 use crate::error::IntegrityVerificationError;
 use reqwest::Client;
 
+pub const SUCCESS_CODE: u16 = 200;
```
Comment (RequescoS) [inline src/api.rs]: NIT: you can use reqwest::StatusCode::OK constant

File: src/performance_measurement.rs [hunk 6]
```diff
@@ -0,0 +1,310 @@
...
+    pub fn inc_failed_requests(&mut self) {
+        self.failed_requests += 1;
+    }
+
+    pub fn add_response_time(&mut self, time: u64) {
+        self.response_time_millis.push(time);
+    }
+
+    pub fn inc_error_code(&mut self, code: u16) {
+        if let Some(count) = self.error_codes.get_mut(&code) {
```
Comment (RequescoS) [inline src/performance_measurement.rs:68]: why get_mut()?

**Comment by RequescoS** [comment]
Great job!

**Review by RequescoS — APPROVED** [review]

**Pull request merged**